content/kotlin/concepts/data-types/data-types.md

```diff
@@ -0,0 +1,390 @@
+
+In simple terms, a byte is a way to represent a small whole number using a computer. It is commonly used to store small pieces of data, such as characters in a text file or colors in an image.
+
+Here is an example of how to declare a **Byte** variable in Kotlin:
+
+```kotlin
+fun main(){
+    val a: Byte = 50
+    val b: Byte = 25
+    val c: Byte = a + b as Byte
+    print(c)
+
+}
+```
+
+**Output** will be :
+
+```
+75
+```
+
+But before using the byte data type make sure you remember the range of the byte data type i.e. -128 to 127.But what if we still cross the range and we are printing Byte then we can check it out with a code example :
+
+```kotlin
+fun main(){
+    val a: Byte = 120
+    val b: Byte = 120
+    val c: Byte = a + b as Byte
+    print(c)
+}
+```
+so here **output** will be :
+```
+-88
+//In this example, the value of c will be -88, because the result of 120 + 120 is 240, which exceeds the valid range for a Byte type.
+```
+
+
+# Char
+
+In Kotlin, the **Char** data type represents a single Unicode character. It is a primitive data type that can hold any character from the Unicode character set, which includes letters, digits, symbols, and special characters.
+
+Here is an example of how to declare a **Char** variable in Kotlin:
+
+```kotlin
+fun main(){
+    val ch: Char = 'A'
+    print(ch)
+}
+```
+
+**Output** will be :
+
+```
+A
+```
+
+In the above example, the ch variable is assigned the value 'A', which is a character from the Unicode character set.
+
+
+# Double 
+
+In Kotlin, the **Double** data type represents a 64-bit floating-point number. It is a primitive data type that can hold values with a fractional component, such as 3.14 or 0.5.
+
+Here is an example of how to declare a **Double** variable in Kotlin:
+
+```kotlin 
+fun main(){
+    val a: Double = 3.14
+    val b: Double = 2.71
+    val c: Double = a * b as Double 
+    print(c)
+}
+```
+
+The **Output** of the above code will be :
+
+```
+8.5094
+```
+In this example, the c variable will be assigned the value 8.5014, which is the result of 3.14 * 2.71.
+
+We can also use the Double data type to perform more advanced operations, such as calculating the square root of a number or raising a number to a power in kotlin. For example:
+
+```kotlin
+
+import kotlin.math.sqrt
+import kotlin.math.pow
+fun main(){
+
+
+val a: Double = 16.0
+val b: Double = sqrt(a)
+val c: Double = a.pow(2)
+
+println("The square root of $a is $b.")
+println("$a raised to the power of 2 is $c.")
+
+}
+```
+
+The **output** will be :
+
+```
+The square root of 16.0 is 4.0.
+16.0 raised to the power of 2 is 256.0.
+```
+
+
+# Float
+
+In Kotlin, the **Float** data type represents a 32-bit floating-point number. It is a primitive data type that can hold values with a fractional component, such as 3.14 or 0.5.
```

Comment:
```suggestion
In Kotlin, the `Float` data type represents a 32-bit floating-point number. It is a primitive data type that can hold values with a fractional component, such as 3.14 or 0.5.
```

This and the above should specify the difference in precision between a float and a double.